fhir2dataset/api_caller.py

```diff
@@ -60,32 +61,34 @@ def get_response(self, url: str):
         Arguments:
             url {str} -- url of the request
         """
-        # TODO : optimize count and paging
-        # count = self._get_count(url)
-        # if count == 0:
-        #     logger.warning(f"there is 0 matching resources for {url}")
-        # if url[-1] == "?":
-        #     url = f"{url}_count={count}"
-        # else:
-        #     url = f"{url}&_count={count}"
-        response = self._get_bundle_response(url)
-        self.status_code = response.status_code
-        try:
-            self.results = response.json()["entry"]
-        except KeyError as e:
-            logger.info(f"Got a KeyError - There's no {e} key in the json data we received.")
-            logger.debug(f"status code of KeyError response:\n{response.status_code}")
-            logger.debug(f"content of the KeyError response:\n{response.content}")
-        try:
-            self.next_url = None
-            for relation in response.json()["link"]:
-                if relation["relation"] == "next":
-                    self.next_url = relation["url"]
-                    break
-        except KeyError as e:
-            logger.info(f"Got a KeyError - There's no {e} key in the json data we received.")
-            logger.debug(f"status code of KeyError response:\n{response.status_code}")
-            logger.debug(f"content of the KeyError response:\n{response.content}")
+        self.next_url = None
+        if self.total is None:
+            self.total = self._get_count(url)
+            logger.info(f"there is {self.total} matching resources for {url}")
+        if self.total != 0:
+            response = self._get_bundle_response(url)
+            self.status_code = response.status_code
+            if "entry" in response.json():
+                # if no resources match the request, self.results = None
+                try:
+                    self.results = response.json()["entry"]
+                except KeyError as e:
+                    logger.info(
+                        f"Got a KeyError - There's no {e} key in the json data we received."
+                    )
+                    logger.debug(f"status code of KeyError response:\n{response.status_code}")
+                    logger.debug(f"content of the KeyError response:\n{response.content}")
+                try:
```

Comment:
you can avoid the try/except here as well:
```python
link = response.json().get("link")
if link and len(link) > 0:
    next_page = link[0]["url"] if link[0]["relation"] == "next" else None
else:
    logger.info("no link information in bundle")
```

Reply:
Also, same as above, `next_url` should not be an attribute of the class

Reply:
Yes thank you !